src/LowerParallelTasks.cpp

```diff
@@ -379,33 +297,23 @@ struct LowerParallelTasks : public IRMutator {
                                                                      closure_arg_index, t.body, target));
 
             if (use_parallel_for) {
-                std::vector<Expr> function_decl_args(3);
-                function_decl_args[0] = make_zero(type_of<void *>());
-                function_decl_args[1] = make_zero(Int(32));
-                function_decl_args[2] = make_zero(type_of<uint8_t *>());
-                Expr function_decl_call = Call::make(Int(32), new_function_name, function_decl_args, Call::Extern);
-
                 std::vector<Expr> args(4);
-                // CodeGen adds user_context for us apparently.
-                //                args[0] = Call::make(type_of<void *>(), Call::get_user_context, {}, Call::PureIntrinsic);
-                args[0] = Call::make(type_of<const void *>(), Call::resolve_function_name, {function_decl_call}, Call::PureIntrinsic);
+                // Codegen will add user_context for us
+
+                // Prefix the function name with "::" as we would in C to make
+                // it clear we're talking about something in global scope in
+                // case some joker names an intermediate Func or Var the same
+                // name as the pipeline.
+                args[0] = Variable::make(Handle(), "::" + new_function_name);
```

Comment:
We created `new_function_name` via a call to  `unique_name` above, so shouldn't this be unnecessary?

EDIT: hm, I guess `c_print_name` could transmute it back into a non-unique name via $->_ transforms? Ugh.